codalab/model/bundle_model.py

```diff
@@ -2180,31 +2180,205 @@ def get_user(self, user_id=None, username=None, check_active=True):
         :param usernames: usernames or emails of users to fetch
         :return: list of matching User objects
         """
         clauses = []
+        offset = 0
+        format_func = None
+        limit = SEARCH_RESULTS_LIMIT
+        count = False
+        sort_key = [None]
+        aux_fields = []  # Fields (e.g., sorting) that we need to include in the query
+
+        # Number nested subqueries
+        subquery_index = [0]
+
+        def alias(clause):
+            subquery_index[0] += 1
+            return clause.alias('q' + str(subquery_index[0]))
+
+        def is_numeric(key):
+            return key in (
+                'id',
+                'time_quota',
+                'parallel_run_quota',
+                'time_used',
+                'disk_quota',
+                'disk_used',
+            )
+
+        def make_condition(key, field, value):
+            # Special
+            if value == '.sort':
+                aux_fields.append(field)
+                if is_numeric(key):
+                    field = field * 1
+                sort_key[0] = field
+            elif value == '.sort-':
+                aux_fields.append(field)
+                if is_numeric(key):
+                    field = field * 1
+                sort_key[0] = desc(field)
+            else:
+                # Ordinary value
+                if isinstance(value, list):
+                    return field.in_(value)
+                if '%' in value:
+                    return field.like(value)
+                return field == value
+            return None
+
         if check_active:
             clauses.append(cl_user.c.is_active)
         if user_ids is not None:
             clauses.append(cl_user.c.user_id.in_(user_ids))
         if usernames is not None:
             clauses.append(or_(cl_user.c.user_name.in_(usernames), cl_user.c.email.in_(usernames)))
+        if keywords is not None:
+            for keyword in keywords:
+                keyword = keyword.replace('.*', '%')
+                # Sugar
+                if keyword == '.count':
+                    count = True
+                    limit = None
+                    continue
+                elif keyword == '.last':
+                    keyword = 'id=.sort-'
+
+                m = SEARCH_KEYWORD_REGEX.match(keyword)  # key=value
+                if m:
+                    key, value = m.group(1), m.group(2)
+                    if ',' in value:  # value is value1,value2
+                        value = value.split(',')
+                else:
+                    key, value = '', keyword
+
+                clause = None
+                # Special functions
+                if key == '.offset':
+                    offset = int(value)
+                elif key == '.limit':
+                    limit = int(value)
+                elif key == '.format':
+                    format_func = value
+                # Bundle fields
+                elif key == 'id':
```

Comment:
I think we can refactor many of these if / elif statements by just using something like `make_condition(key, getattr(cl_user.c, key), value)`